sdk/ai/azure-ai-inference/azure/ai/inference/aio/_patch.py

```diff
@@ -132,8 +132,158 @@ class ChatCompletionsClient(ChatCompletionsClientGenerated):
     def __init__(
         self, endpoint: str, credential: Union[AzureKeyCredential, "AsyncTokenCredential"], **kwargs: Any
     ) -> None:
-        self._model_info: Optional[_models.ModelInfo] = None
-        super().__init__(endpoint=endpoint, credential=credential, **kwargs)
+        # First pop out any unlisted embedding settings, since we need to remove them before saving kwargs
+        model_info: Optional[_models.ModelInfo] = kwargs.pop("model_info", None)
+        model_extras: Optional[Dict[str, Any]] = kwargs.pop("model_extras", None)
+        frequency_penalty: Optional[float] = kwargs.pop("frequency_penalty", None)
+        presence_penalty: Optional[float] = kwargs.pop("presence_penalty", None)
+        temperature: Optional[float] = kwargs.pop("temperature", None)
+        top_p: Optional[float] = kwargs.pop("top_p", None)
+        max_tokens: Optional[int] = kwargs.pop("max_tokens", None)
+        response_format: Optional[Union[str, _models.ChatCompletionsResponseFormat]] = kwargs.pop(
+            "response_format", None
+        )
+        stop: Optional[List[str]] = kwargs.pop("stop", None)
+        tools: Optional[List[_models.ChatCompletionsToolDefinition]] = kwargs.pop("tools", None)
+        tool_choice: Optional[
+            Union[str, _models.ChatCompletionsToolSelectionPreset, _models.ChatCompletionsNamedToolSelection]
+        ] = kwargs.pop("tool_choice", None)
+        seed: Optional[int] = kwargs.pop("seed", None)
+        model: Optional[str] = kwargs.pop("model", None)
+
+        # Remember all listed input arguments, as we may need to re-apply them when a new client is
+        # construced in the `with_defaults` method
+        self._endpoint = endpoint
+        self._credential = credential
+        self._init_kwargs = kwargs
+
+        # Store default embedding settings, to be applied in all future service calls
+        self._model_info = model_info
+        self._model_extras = model_extras
+        self._frequency_penalty = frequency_penalty
+        self._presence_penalty = presence_penalty
+        self._temperature = temperature
+        self._top_p = top_p
+        self._max_tokens = max_tokens
+        self._response_format = response_format
+        self._stop = stop
+        self._tools = tools
+        self._tool_choice = tool_choice
+        self._seed = seed
+        self._model = model
+
+        super().__init__(endpoint, credential, **kwargs)
+
+    def with_defaults(
```

Comment:
The ownership of the transport is not correct:

```python
import asyncio

import azure.core.pipeline.transport
import azure.core.credentials as credentials
import azure.ai.inference.aio as inference

async def run():
    transport = azure.core.pipeline.transport.AioHttpTransport()
    async with inference.ChatCompletionsClient(endpoint='https://some.com', credential=credentials.AzureKeyCredential("123"), transport=transport) as client:
        async with client.with_defaults(seed=12) as defaulted_client:
            try:
                print("inner")
                await defaulted_client.get_model_info()
            except Exception as e:
                print(e)
        try:
            print("outer")
            await client.get_model_info()
        except Exception as e:
            print(e)
                
asyncio.run(run())
```

Reply:
Thank you @johanste for the code! I got a repro and I understand the issue now. Not clear to me how to resolve it. I'll continue looking tomorrow.

Reply:
@johanste resolving this since we decided not to ship a `with_defaults` method, but instead add chat completions/embedding settings in the constructor.